pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -152,6 +152,7 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
     //
     Character delim = ',';
     arrayOidToDelimiter.put(oid, delim);
+    arrayOidToDelimiter.put(arrayOid, delim);
```

Comment:
The `arrayOidToDelimiter` map should be initialized to 2x size to allow for these additional entries

Reply:
good catch...

Reply:
You mean in the constructor, or somewhere else? It doesn't seem to have a size there to begin with (not that it wouldn't be good to add one there I suppose).

Reply:
If you'd like to see `arrayOidToDelimeter = new HashMap(types.length * 2)` in the constructor, I can do that. But would it make sense to do the same with `pgNameToOid`, `pgNameToJavaClass`, `pgNameToPgObject`, and `types.length * 1` for `pgArrayToPgType`?

Also, should we consider updating `getArrayDelimiter` to fall back to `arrayOidToDelimiter.get(pgArrayToPgType.get(oid))` before hitting the database?

Again, I'm not sure what the intended behavior is for getting delimiters for the array-type OID instead of the underlying type in the first place; I'd expected `getArrayDelimiter` to be receiving the OID for `int8` and not its array companion when the field is an `int8[]`, and the OID for `int8[]` in multidimensional arrays. But even **if** that's correct, it's a larger change than is probably worth doing here.

Reply:
> If you'd like to see `arrayOidToDelimeter = new HashMap(types.length * 2)` in the constructor, I can do that. But would it make sense to do the same with `pgNameToOid`, `pgNameToJavaClass`, `pgNameToPgObject`, and `types.length * 1` for `pgArrayToPgType`?

I would vote `arrayOidToDelimeter` to be `types.length * 2.5` and the rest to be `types.length * 1.5`. The `PgConnection` class also adds some values every time.

> Also, should we consider updating `getArrayDelimiter` to fall back to `arrayOidToDelimiter.get(pgArrayToPgType.get(oid))` before hitting the database?

I am somewhat ambivalent here. I think the currently proposed change to populate the map for these core types would appear to cover the vast majority of cases. Adding this would avoid a db query per connection lifetime for non-core types?

> I'd expected `getArrayDelimiter` to be receiving the OID for `int8` and not its array companion when the field is an `int8[]`, and the OID for `int8[]` in multidimensional arrays.

As I recall, this is driven by what direction we are going (inserting vs. retrieving). On inserts we are primarily starting with the base type `int8` whether single dimension or multi-dimensional. On retrieves I seem to recall that the column type is described as `int8[]`, again regardless of single or multi-dimensional.

Reply:
Thanks for the feedback and explanation. I've updated per your comments in c0921a0. I agree that there's probably no need to fall back to `arrayOidToDelimiter.get(pgArrayToPgType.get(oid))` if we're doing everything else right.

Working through `addCoreType` it looks like the other arrays are added to twice - once for `pgTypeName` and again for `pgArrayTypeName` (eg. `int8` and `_int8`, respectively). That's why I suggested `types.length * 2` for the other arrays.